Packs/PagerDuty/Integrations/PagerDuty/PagerDuty.py

```diff
@@ -553,31 +551,96 @@ def fetch_incidents():
 
 
-def configure_status(status='triggered,acknowledged'):
+def configure_status(status='triggered,acknowledged') -> str:
     statuses = status.split(',')
-    statuses_string = "&" + STATUSES + '='
+    statuses_string = f"&{STATUSES}="
     statuses = statuses_string.join(statuses)
-    status_request = '&' + STATUSES + '=' + statuses
+    status_request = f'&{STATUSES}={statuses}'
 
     status_request = status_request + INCLUDED_FIELDS + UTC_PARAM
     return status_request
 
 
-def get_incidents_command(since=None, until=None, status='triggered,acknowledged', sortBy=None, incident_key=None):
+def pagination_incidents(param_dict: dict, url: str) -> list[dict]:
+    """
+    Retrieves incident data through paginated requests.
+
+    Args:
+        param_dict (dict): A dictionary containing parameters for controlling pagination, including:
+            - 'page': Current page number (optional).
+            - 'page_size': Number of incidents per page (optional).
+            - 'limit': Maximum number of incidents to retrieve (optional).
+            - Additional parameters to include in the API request.
+
+        url (str): The URL of the API endpoint for incident data.
+
+    Returns:
+        list[dict]: A list of dictionaries, where each dictionary represents an incident.
+
+    Notes:
+        This function supports pagination for efficient retrieval of large datasets. It calculates
+        the appropriate 'limit' and 'offset' values based on the provided arguments.
+
+    Examples:
+        To retrieve incidents for the second page with a page size of 20:
+        >>> pagination_incidents({'page': 2, 'page_size': 20}, 'https://api.example.com/incidents', {})
+
+        To retrieve the first 50 incidents without explicit pagination:
+        >>> pagination_incidents({'limit': 50}, 'https://api.example.com/incidents', {'status': 'open'})
+    """
+    def _get_page(**pagination_args) -> list[dict]:
+        # param_dict must be before pagination_args for merging to work correctly
+        return http_request('GET', url, param_dict | pagination_args).get("incidents", [{}])
+
+    pages: list = []
+
+    page = arg_to_number(param_dict.get("page"))
+    page_size = arg_to_number(param_dict.get("page_size"))
+    if page is not None and page_size is not None:
+        limit = page_size
+        offset = (page - 1) * page_size
+
+    else:
+        limit = arg_to_number(param_dict.get("limit")) or 50
+        offset = 0
+
+    if limit > INCIDENT_API_LIMIT:
+        for offset in range(0, limit - INCIDENT_API_LIMIT, INCIDENT_API_LIMIT):
+            pages += _get_page(
+                limit=INCIDENT_API_LIMIT,
+                offset=offset)
+
+        # the remaining call can be less than OR equal the api_limit but not empty
+        limit = limit % INCIDENT_API_LIMIT or INCIDENT_API_LIMIT
+        offset += INCIDENT_API_LIMIT
+
+    pages += _get_page(
+        limit=limit,
+        offset=offset)
+
+    return pages
```

Comment:
naming correction -  in the end this function returns a page, not pages. even if inside it may collect smaller pages, but its building one page.